### ISSUE TITLE
Not building with Visual Studio 2019 Preview and /std:c++latest

### ISSUE THREAD

**Issue by IanMDay** [electronicarts/EASTL#312]
Many build errors with the latest VS 16.3.0 Previews. Too many to list, but most are to do with char8_t with /std:c++latest compiler option. This is just a plain build of EASTL standalone freshly cloned and new CMake generation.

A single example from EAMain.cpp

`EA::StdC::Vsnprintf(pBuffer, nReturnValue + 1, pFormat, arguments);`

```
error C2665: 'EA::StdC::Vsnprintf': none of the 4 overloads could convert all the argument types
10>C:\git\lib\EASTL\test\packages\EAStdC\include\EAStdC/EASprintf.h(201,18) : message : could be 'int EA::StdC::Vsnprintf(wchar_t *__restrict ,size_t,const wchar_t *__restrict ,va_list)' (compiling source file C:\git\lib\EASTL\test\packages\EAMain\source\EAMain.cpp)
10>C:\git\lib\EASTL\test\packages\EAStdC\include\EAStdC/EASprintf.h(193,17) : message : or       'int EA::StdC::Vsnprintf(char32_t *__restrict ,size_t,const char32_t *__restrict ,va_list)' (compiling source file C:\git\lib\EASTL\test\packages\EAMain\source\EAMain.cpp)
10>C:\git\lib\EASTL\test\packages\EAStdC\include\EAStdC/EASprintf.h(186,17) : message : or       'int EA::StdC::Vsnprintf(char16_t *__restrict ,size_t,const char16_t *__restrict ,va_list)' (compiling source file C:\git\lib\EASTL\test\packages\EAMain\source\EAMain.cpp)
10>C:\git\lib\EASTL\test\packages\EAStdC\include\EAStdC/EASprintf.h(178,17) : message : or       'int EA::StdC::Vsnprintf(char *__restrict ,size_t,const char *__restrict ,va_list)' (compiling source file C:\git\lib\EASTL\test\packages\EAMain\source\EAMain.cpp)
10>C:\git\lib\EASTL\test\packages\EAMain\source\EAMain.cpp(311,94) : message : while trying to match the argument list '(char *, int, const char8_t *, va_list)'
```

These are calling external library routines that all take const char* equivelants which the compiler rejects.

**Comment by rparolin** [comment]
Have you tried with the latest release?  It contains char8_t fixes for VS2019.

**Comment by IanMDay** [comment]
I did yes. Just cloned the latest. Has a lot of char8_t errors when using latest with 2019 preview.

**Comment by wmamrak** [comment]
I think that until the fix arrives, the C++ language standard should be downgraded from latest to 17.

**Comment by rparolin** [comment]
I'll add a MSVC latest as soon as Travis CI supports it.

**Comment by rparolin** [comment]
I have fixes locally for this and it will be in the next EASTL release.

**Comment by IanMDay** [comment]
That's great. Any movement on the submodule issue? Going to be difficult to integrate these otherwise.

**Comment by rparolin** [comment]
No.  Don't have a better alternative yet than using submodules.

**Comment by rparolin** [comment]
This issue should be resolved with 3.15.00.

**Comment by IanMDay** [comment]
Still fails with quite a lot for me with 3.15.
